components.go

```diff
@@ -89,6 +89,18 @@ func (cb *ComponentBase) AddProperty(property ComponentProperty, value string, p
 	cb.Properties = append(cb.Properties, r)
 }
 
+// RemoveProperty removes from the component all properties that has
+// the name passed in removeProp.
+func (cb *ComponentBase) RemoveProperty(removeProp ComponentProperty) {
+	var keptProperties []IANAProperty
+	for i := range cb.Properties {
+		if cb.Properties[i].IANAToken != string(removeProp) {
+			keptProperties = append(keptProperties, cb.Properties[i])
```

Comment:
I really prefer readability over premature optimization. For the above in-place attempt, I really have to think very carefully about its workings, and whether it might have some edge cases or such. My code is really straightforward to read and understand. Is making a new array really an issue performance wise here?

Reply:
It isn't a core concern of the library, as I found another area where I preference memory over performance.

I guess the real issue here is if people are using the slice directly. For simplicity I guess it's better to go with yours for now. 

However there are lint issues remaining if you resync it should help.

One thing your version will do is put a nil slice instead of an empty slice. Which is different, but I don't know if it's a concern:

https://go.dev/play/p/iRgj5Jsqe8O